node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/servicedump/VespaServiceDumperImpl.java

```diff
@@ -0,0 +1,170 @@
+
+    public VespaServiceDumperImpl(ContainerOperations container, SyncClient syncClient, NodeRepository nodeRepository) {
+        this.container = container;
+        this.syncClient = syncClient;
+        this.nodeRepository = nodeRepository;
+    }
+
+    @Override
+    public void processServiceDumpRequest(NodeAgentContext context) {
+        Instant startedAt = Instant.now();
+        NodeSpec nodeSpec = context.node();
+        ServiceDumpReport request = nodeSpec.reports().getReport(ServiceDumpReport.REPORT_ID, ServiceDumpReport.class)
+                .orElse(null);
+        if (request == null || request.isCompletedOrFailed()) {
+            context.log(log, Level.FINE, "No service dump requested or dump already completed/failed");
+            return;
+        }
+        if (isNullTimestamp(request.getCreatedMillisOrNull())) {
+            handleFailure(context, request, startedAt, null, "'createdMillis' is missing or null");
+            return;
+        }
+        String configId = request.configId();
+        if (configId == null) {
+            handleFailure(context, request, startedAt, null, "Service config id is missing from request");
+            return;
+        }
+        Instant expiry = expireAt(startedAt, request);
+        if (expiry.isBefore(startedAt)) {
+            handleFailure(context, request, startedAt, null, "Request already expired");
+            return;
+        }
+        try {
+            context.log(log, Level.FINE,
+                    "Creating dump for " + configId + " requested at " + Instant.ofEpochMilli(request.getCreatedMillisOrNull()));
+            storeReport(context, createStartedReport(request, startedAt));
+            Path directoryInNode = context.pathInNodeUnderVespaHome("tmp/vespa-service-dump");
+            Path directoryOnHost = context.pathOnHostFromPathInNode(directoryInNode);
+            Files.deleteIfExists(directoryOnHost);
+            Files.createDirectory(directoryOnHost);
+            Path vespaJvmDumper = context.pathInNodeUnderVespaHome("bin/vespa-jvm-dumper");
+            CommandResult result = container.executeCommandInContainerAsRoot(
+                    context, vespaJvmDumper.toString(), configId, directoryInNode.toString());
+            context.log(log, Level.FINE, "vespa-jvm-dumper exit code: " + result.getExitCode());
+            context.log(log, Level.FINE, "vespa-jvm-dumper output: " + result.getOutput());
+            if (result.getExitCode() > 0) {
+                handleFailure(context, request, startedAt, null, "Failed to create dump: " + result.getOutput());
+                return;
+            }
+            URI destination = serviceDumpDestination(nodeSpec, createDumpId(request));
+            context.log(log, Level.FINE, "Uploading files with destination " + destination + " and expiry " + expiry);
+            List<SyncFileInfo> files = dumpFiles(directoryOnHost, destination, expiry);
+            logFilesToUpload(context, files);
```

Comment:
We should log the files being uploaded at Level.INFO.  And it would be nice if the message would contain the actual list of files being uploaded, as sync() may choose to skip some.